All Updates so far

FominSergiy opened the pull request:
Working in a dev brunch on this up.

Progress made so far
- Got base components for the board, squares
- Set up reducers to run with thunks
- Attached moving to buttons and arrow buttons on keyboard

Left To Do:
- set up timer for 60 seconds
- overall game styling
- update my file structure (just looks bad)

Commit:
created project with a base react components and base redux logic to display a 5x5 grid for the game

Commit:
set the init position of the mage for the game and added rendering for the square

Commit:
adding formats for button controls and a background image

Commit:
adding button components, styling and removing old unused zip file

Commit:
abstracted move func to be applicable to 4 buttons, and adjusted reducer to take the actions

Commit:
updated buttons to only care about the state items needed to make a move

Commit:
feature - set up boudnaries and game over conditions. updated index.js , utils, and reducer files

Commit:
added random coin generator

Commit:
abstracted getting image in a square into a function

Commit:
added reducer to manage score

Commit:
refactored currentPos reducer to hold positions of both, the coin and the mage. As such updated the rest of the code base accordingly

Commit:
updated all reducers to include a single reset method, which resets states in all slices

Commit:
have a working version of scoring system and coin rendering when mage and coin overlap -aka a score

Commit:
added event listeners for buttons. having bugs here, but can't see the reason why

Commit:
added actions.js file to store action creators - will replace regular dispatches with action creator functions

Commit:
updated a bug that turned coin into mage when a coin was generated on a prevPos for mage

Commit:
updated filepath for the coin

Commit:
added redux-thunk and refactored makeMove func to run off of the Action creators, which are inside actions.js

Commit:
added redux logger for debugging and redux-thunk to update state with thunks

Commit:
new const reference breaks when using gh pages. reverting back

Commit:
added game timer for 60 seconds

Commit:
removed console.log()s

 On branch dev
 Changes to be committed:
	modified:   src/features/button-controls/buttonUtils.js
	modified:   src/index.js
	modified:   src/reducer.js

FominSergiy commented:
Left To Do:

overall game styling
update my file structure (just looks bad)

Commit:
added timer prop to buttons

karlowski commented on package.json:
```diff
@@ -1,5 +1,6 @@
 {
-  "name": "tanks_game",
+  "homepage": "http://FominSergiy.github.io/dodgeFairy",
+  "name": "dodge_fairy",
```
from tanks to fairies! :D

FominSergiy replied:
right ahaha

Commit:
added more styling around scores. Added a div to contain rules and added messages when the game is over

Commit:
cleaned up index.js

Commit:
added an image and text styling for the game over screen

Commit:
moved gameOver to a separate React component to add event listener to it to be able to reset the game with Enter key

Commit:
removed button controls. will add them back once working on a mobile version

	modified:   src/features/game-over/gameOver.js
	modified:   src/utils/utils.js

Commit:
changed app icons and modified html head of the document

Commit:
updated gh-pages url

Commit:
moved board component into features

Commit:
removed unnecessary computed prop key

Commit:
added APIs and components for the scorebaord, including thunks. able to read results. will need to create a table structure to display them

Commit:
added function to sort incoming scores and store then in an array in object: {user: foo, score: bar}

Commit:
added axios to work with requests

Commit:
added function to return an array of Score components back to the scoreboard

Commit:
added styling for the scoreboard

Commit:
added styling to restart button

Commit:
moved redux components into its own folder

Commit:
updated redux thunk function to process the records before they are saved to redux store. that way appended extraction of top scores as an array. will need it to set new top scores on the scoreboard

Commit:
added a function to calculate new record score, depending on the size of the scorebaord

Commit:
added docstring to new function

	modified:   src/features/scoreboard/scoreboardUtils.js

	modified:   src/features/scoreboard/scoreboardUtils.js

Commit:
added a reducer to be used to manage state for when a new record is being set

Commit:
added raw styling for the new record screen

Commit:
added reducer and an action to store userName state from the form, when user enters their name

Commit:
added get request to add new scores. Very messy at the moment. Need to work on the component re-render. Ideally once a new entry is created, scoreboard should re-render

Commit:
fixed a big - score did not record if it was the same as the one currently on the leaderboard

Commit:
updated enter keydown. Now when a new record is set keydown will be controlled from the Form element and will be responsible for the form submission. if it's not a new score, then keydown on 13 will reset the game

Commit:
updated the topScores reducer to have null as the init value, and changed the score calc function.Fixed a bug where a new record score would be set because the api call has not yet returned any data

Commit:
updated components to re-rende the form once a new score has been submitted

Commit:
fixed bug with adding a score. did not extract variables from the return array properly

Commit:
fixed the issue with new scores re-render. The last issue is to fix the problem when a new top score won't render because all the scores are the same

Commit:
fixed the issue with min score not rendering. added a different calc to the function responsible for calculating a winner score

Commit:
table re-render is achieved here when a new score is submitted and the scoreboard needs a re-render

Commit:
added fix to handle addition of a new top score

Commit:
TEST - trying to fix a bug with multiple key-presses causing duplication. Trying to fix this

	modified:   src/constants.js
	modified:   src/features/board/board.js
	modified:   src/features/button-controls/buttonUtils.js
	modified:   src/features/scoreboard/scoreboard.js
	modified:   src/features/square/square.js
	modified:   src/index.js
	modified:   src/redux/actions.js
	modified:   src/redux/reducers.js
	modified:   src/redux/store.js

Commit:
fixed the issue with bug from multiple simultaneous keyDown events

Commit:
removed scores array from the arr dependancies on useEffect hook to avoid unnecesary re-render when new score is received and the scores array is re-rendered

Commit:
added error handling to my api requests

Commit:
moved api end-points to an env file. PREV END POINTS ARE DEV

Commit:
moved endpoints to env file and updated npm deploy command. APIs still visible (I know) but access to them is restricted from within the Azure

Commit:
updated file names and intructions

karlowski commented on src/features/game-over/gameOver.js:
```diff
@@ -0,0 +1,46 @@
+import { useDispatch, useSelector } from "react-redux";
+import React from 'react';
+
+export const GameOver = (props) => {
```
looks very nice when even *game over stuff* separated in different module

karlowski commented on src/utils/utils.js:
```diff
@@ -0,0 +1,165 @@
+    if (score > 75) msg='TO THE MOOOON!';
+
+    return msg;
+}
```
what  is separated utils file for btw?
kinda constants but for more complex stuff like functions..?

FominSergiy replied:
I don't even know. I created it initially to hold all the utills for the project, but then it made sense, for some pieces, to hold utils in the same place as the code for the component.. I really need to re-think this
